services/structureddate/structureddate/src/main/java/org/collectionspace/services/structureddate/antlr/ANTLRStructuredDateEvaluator.java

```diff
@@ -220,6 +221,26 @@ public void exitUncertainDate(UncertainDateContext ctx) {
 		int earliestInterval = DateUtils.getCircaIntervalYears(earliestDate.getYear(), earliestDate.getEra());
 		int latestInterval = DateUtils.getCircaIntervalYears(latestDate.getYear(), latestDate.getEra());
 
+		if (earliestDate.getEra() == Era.BCE && latestDate.getEra() == Era.BCE) {
+			// Improved precision for BC dates
+			if ((latestDate.getMonth() == 12 && latestDate.getDay() == 31) &&
+				(earliestDate.getMonth() == 1 && earliestDate.getDay() == 1)) {
```

Comment:
Why do you need to check if both earliest and latest are BC, and the month and day? Is there a case where you don't get the expected result by moving this algorithm into getCircaIntervalYears?